karate-core/src/main/java/com/intuit/karate/driver/DevToolsDriver.java

```diff
@@ -964,5 +967,11 @@ public void inputFile(String locator, String... relativePaths) {
         Integer nodeId = elementId(locator);
         method("DOM.setFileInputFiles").param("files", files).param("nodeId", nodeId).send();
     }
-    
+
+    public Object scriptAwaitPromise(String expression) {
```

Comment:
I'm going to add this method to karate now, apologies for stealing your code, but just want to make a release

Reply:
cool!

Reply:
made changes with new commit. Thank you.